### DISCUSSION TITLE
Introduce $slotVariables to class-based components

### DISCUSSION THREAD

**Discussion by Pennywise96** [laravel/framework#52955] [Ideas]
I’ve been experimenting with Blade’s class-based components and made some enhancements.

In this repository, I introduced `$slotVariables` to a class-based component. This allows developers to specify which public properties or methods should be passed to the slot.

You can explore the changes in the repo: https://github.com/Pennywise96/slotVariables.

I’ve already created a route and view, `/test`, with a `<x-posts>` component for testing.

### How it looks in the view:

```
<x-posts>
    @foreach($posts as $post)
        <li>{{ $post }}</li>
    @endforeach
</x-posts>
```

### Changes made:
`Component.php`

- Added a new property: `$slotVariables`
- Added a new method: `slotData()`

`CompilesComponents.php`

- Modified `public static function compileClassComponentOpening()` to inject and extract `slotData()`
- Updated `public function compileEndComponentClass(`) to unset the injected data

I believe the code can be further improved, and I’d appreciate your feedback on how to make it more elegant.
What do you think?

**Comment by Sophist-UK** [comment]
I would love to review the changes you made.

Unfortunately you didn't start by creating a baseline in Github (or by forking Laravel) and then adding your changes as a separate commit where I can see the diff. And I don't have time to work out what changes were made in the files your refer to.

**Reply by Pennywise96** [reply]
I've now forked Laravel and pushed a commit titled "WIP". You can check it out here: [GitHub - Pennywise96/laravel-slot-variables](https://github.com/Pennywise96/laravel-slot-variables).

**Reply by Sophist-UK** [reply]
The code style looks fine.

I have not tested it, however I think that in `CompilesComponents.php` you should probably test for slots usage and only include the extra code if they are being used.
